org.adempiere.base/src/org/compiere/model/PO.java

```diff
@@ -1394,12 +1418,34 @@ protected boolean load (String uuID,String trxName)
 		{
 			if (i != 0)
 				sql.append(",");
-			String columnSQL = p_info.getColumnSQL(i);
-			if (!p_info.isVirtualColumn(i))
+			String columnSQL;
+			if (p_info.isVirtualColumn(i))
 			{
+				boolean lazyLoad = true;
+				if(virtualColumns!=null)
+				{
+					for(String virtualColumn : virtualColumns)
+					{
+						if(p_info.getColumnName(i).equalsIgnoreCase(virtualColumn))
+						{
+							lazyLoad = false;
+							break;
+						}
+					}
+				}
+
+				if(lazyLoad)
+					// initialize with NULL - values will be lazy-loaded by getters
+					columnSQL = "NULL AS " + DB.getDatabase().quoteColumnName(p_info.getColumnName(i));
+				else
+					columnSQL = p_info.getColumnSQL(i);
```

Comment:
needs to update loadedVirtualColumns here

Reply:
Not really, the set is updated automatically whenever there's data for these columns. This is performed at `PO.loadColumn(ResultSet, int)`.

Reply:
hmm ... can you clarify how would this work if the virtual column return a null value ?

Reply:
We've found a cleaner (and more efficient) approach to do this, without `Null.NULL`.  Virtual columns whose queries return `NULL` are treated no different than then rest.